java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -231,6 +235,8 @@ private void readVector(Field field, FieldVector vector) throws JsonParseExcepti
         throw new IllegalArgumentException("Expected field " + field.getName() + " but got " + name);
       }
       int count = readNextField("count", Integer.class);
+      vector.allocateNew();
```

Comment:
I am still not very familiar with when to call the allocateNew() method. Why is it necessary to set the value count?

Reply:
`allocateNew` allocates the buffer for the vector.
`setValueCount` sets the number of values in the vector.

We need to call `setValueCount` to correctly create the vector

Reply:
I think we should set lastSet before setting the value count else setValueCount() will corrupt the vector.

Reply:
Unfortunately it's hard to do it here unless we do some crazy instance matching here, I don't want to do that because it's too hard to maintain.

The correct way I think is to use the proper `loadFieldBuffers` in this class which initialize the vectors correctly and set things like `lastSet`( The TODO above)

Reply:
ok, sounds good to me. I think we need lastSet only for VarChar, VarBinary and ListVector?

Reply:
I am not too sure, but at least, + their Nullable version.